### ISSUE TITLE
[firestore-Translate Text] Hindi as a source language isn't getting translated

### ISSUE THREAD

**Issue by itssidhere** [firebase/extensions#659]
### [READ] Step 1: Are you in the right place?

Issues filed here should be about bugs for a **specific extension in this repository**.
If you have a general question, need help debugging, or fall into some
other category use one of these other channels:

- For general technical questions, post a question on [StackOverflow](http://stackoverflow.com/)
  with the firebase tag.
- For general Firebase discussion, use the [firebase-talk](https://groups.google.com/forum/#!forum/firebase-talk)
  google group.
- To file a bug against the Firebase Extensions platform, or for an issue affecting multiple extensions, please reach out to
  [Firebase support](https://firebase.google.com/support/troubleshooter/contact/) directly.

### [REQUIRED] Step 2: Describe your configuration

- Extension name: **\_** firestore-translate-text
- Extension version: **\_**0.1.5
- Configuration values (redact info where appropriate):
  - **\_** Target languages - en,es,de,fr,hi,tl
  - **\_**

### [REQUIRED] Step 3: Describe the problem
When I am setting the text to be translated in Hindi, the text isn't getting translated. Though it works in google translator.

#### Steps to reproduce:

Set the text "Mera naam siddharth hai"

##### Expected result
en: My name is Siddharth

##### Actual result
Mera naam Siddharth hai

**Comment by dackers86** [comment]
hi @itssidhere.

I have managed to successfully replicate the issue. Interestingly, this works as expected through Google Translate...

https://translate.google.com/?sl=hi&tl=en&text=Mera%20naam%20siddharth%20hai&op=translate

Using Hindi scripture also seems to work as expected...

![image](https://user-images.githubusercontent.com/2060661/119501114-77d6ad80-bd60-11eb-9df4-ea82c2096119.png)

but not when using `Mera naam siddharth hai` through the extension.


The extension utilises  the Node Api library, which under the hood calls the `v2` rest Api endpoint. This endpoint returns the same response...

https://translation.googleapis.com/language/translate/v2?key=

Body:

```js
{
  "q": ["Mera naam siddharth hai"],
  "target": "en"
}
```

Response:

```js
{
    "data": {
        "translations": [
            {
                "translatedText": "Mera naam siddharth hai",
                "detectedSourceLanguage": "hi"
            }
        ]
    }
}
```

**Comment by itssidhere** [comment]
@dackers86 This also happens with the ML kit.

**Comment by yamankatby** [comment]
Hi there,
This doesn't seem to be an issue in the extension since the translation API returns the same response.

![capture](https://user-images.githubusercontent.com/35961879/183064431-11a161ae-fae2-4cc9-97dd-c398f05d3c65.png)

TL;DR;

As I understood, the provided example sentence isn't official Hindi, it's something called "[Hindi Latin-Phonetics](https://en.wikipedia.org/wiki/Devanagari#Phonetic)". It's not official and not widely supported, however, some translating services such as translate.google.com (the website) use an extra algorithm called [phonetics converter](https://en.wikipedia.org/wiki/Google_IME) allows non-Latin languages to be written with Latin characters.

For example, translation from Arabic using Latin letters:

![Screen Shot 2022-08-05 at 14 27 41](https://user-images.githubusercontent.com/35961879/183068259-97c1320b-ff1f-42d3-a9c1-de8dc772787a.png)

**Comment by cabljac** [comment]
Closing this as not planned, as changes to the actual translation API are out of scope of this repo.
